### PR TITLE
Remove heretic chaplain spawn

### PR DESCRIPTION
**Pull request by BoundCNapalm**
## About The Pull Request

Adds Chaplain to protected roles in Heretic Smuggler and Heretic.

Haven't tested but it only modifies protected roles, so it should be fine.

## Why It's Good For The Game

Heretic is one of few roles that can spawn on its direct counter, without a real way to stop it thanks to rulings made to enhance the heretic experience. This creates an incredibly powerful antagonist whose only weakness is that no one who spawns as a heretic chaplain ahelps when they are killed for their killer using metaknowledge.

This PR adds JOB_CHAPLAIN to the protected roles list of all Heretic role spawns in dynamic I am aware of. This prevents heretic spawns as Chaplain. Hopefully, this can also be backed up with a policy update relating to denials of all OPFORs for all future Heretic/Cult Chaplains.

## Changelog

All this does is prevent heretic chaplain spawns by adding them to protected roles.

:cl:
code: Two lines of code, Adds JOB_CHAPLAIN to protected roles in the related files.
config: Axes natural heretic chaplain spawns. 
/:cl:

### CODE REVIEW

**Comment by Eastwardrope** [comment]
Why wasn't this a thing by default?

**Comment by BoundCNapalm** [comment]
> Why wasn't this a thing by default?

I don't know. My guess is that it is something that TG hasn't accounted for or just a Skyrat moment. In either circumstance, it wasn't done at first, and TG can't do all the work. Lol.

**Comment by destrucktoid** [comment]
As a avid chaplain player i can't help but say that i dislike this change for many reasons, first amongst which is that chaplain being heretic is an absolutely great place for gimmicks of all kind. Also chaplain's being heretic isn't default ANYWHERE, not on TG, not on Skyrat, not on Fulp. This is in my opinion a change that fixes a problem that doesn't exist. For starters, when do you see normal chaplains validhunting heretics? Never, the most they ever do is tell sec "oh, that's magic." because the admins will bwoink them if they go out of their way to kill actual heretics. Overall this is a change that fixes a non-existent issue and removes one of the best roles for heretic gimmicks.

**Comment by Gr33d-y** [comment]
> As a avid chaplain player i can't help but say that i dislike this change for many reasons, first amongst which is that chaplain being heretic is an absolutely great place for gimmicks of all kind. Also chaplain's being heretic isn't default ANYWHERE, not on TG, not on Skyrat, not on Fulp. This is in my opinion a change that fixes a problem that doesn't exist. For starters, when do you see normal chaplains validhunting heretics? Never, the most they ever do is tell sec "oh, that's magic." because the admins will bwoink them if they go out of their way to kill actual heretics. Overall this is a change that fixes a non-existent issue and removes one of the best roles for heretic gimmicks.

Chaps hunting magic antags is their job, no one is gonna bwoink a chap for doing that. allowing them to be a magic antag is like allowing sec to be traitor.

**Comment by BoundCNapalm** [comment]
Gonna drop a storyteller patch soon specifically to accomplish the purpose of this PR

**Comment by Gr33d-y** [comment]
> > > As a avid chaplain player i can't help but say that i dislike this change for many reasons, first amongst which is that chaplain being heretic is an absolutely great place for gimmicks of all kind. Also chaplain's being heretic isn't default ANYWHERE, not on TG, not on Skyrat, not on Fulp. This is in my opinion a change that fixes a problem that doesn't exist. For starters, when do you see normal chaplains validhunting heretics? Never, the most they ever do is tell sec "oh, that's magic." because the admins will bwoink them if they go out of their way to kill actual heretics. Overall this is a change that fixes a non-existent issue and removes one of the best roles for heretic gimmicks.
> > 
> > 
> > Chaps hunting magic antags is their job, no one is gonna bwoink a chap for doing that. allowing them to be a magic antag is like allowing sec to be traitor.
> 
> Chaplains are NOT security, if they were they'd be given the title of inquisitor and made security. Their job, by description is to "hold services and funerals, cremate people, preach your religion, protect the crew against cults." Your null rod and your abilities are there to help you validhunt CULT not to validhunt heretics. The chaplain's job is to protect against **religious** threats, IE Clockwork cult and blood cult. They already have built in protections against becoming either of those as they cannot become blood cultists and back when clock cult was a thing they couldn't be converted to them either. Also, for the record i HAVE been bwoinked for hunting for heretics as chaplain before on bubberstation (admittedly a LOOONG time ago).

That's just false, straight up. One of chaps jobs is protecting the crew from all magical antags, not just cult. They are resistant to wizard, heretic, Rev, and of course the cults. This is very much by design and on purpose. As you said, you got bwoinked for this when we first started but as it stands they should never have done that at any point in our servers lifespan. Chap being the counter and main force against magic antags is not my opinion, it is quite literally how it is on most servers, including ours. You thinking that's not the case does not change that fact.

**Comment by BoundCNapalm** [comment]
I would prefer if debates on the purpose of the Chaplain remained in the #discuss_anything channel on discord. As it stands, my interpretation of the Chaplain's purpose is being emphasized in this PR, and others agree with me that this is a necessary thing.

Once I update my "Adds Hereprisoner" PR, I will add a patch to Storyteller for this PR to patch out that little exception Storyteller allowed where Heretic Chaplain could spawn

**Comment by zeroisthebiggay** [comment]
> > Why wasn't this a thing by default?
> 
> I don't know. My guess is that it is just a Skyrat moment, or TG coders haven't gotten to it yet. In either circumstance, it wasn't done at first, and TG can't do all the work. Lol.

it's default on citadel

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -208,6 +208,7 @@
 		JOB_HEAD_OF_SECURITY,
 		JOB_PRISONER,
 		JOB_SECURITY_OFFICER,
+		JOB_CHAPLAIN,
```
Comment (nevimer) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm:211]:
```suggestion
		JOB_CHAPLAIN, // BUBBER EDIT ADDITION
```

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -214,6 +214,7 @@ GLOBAL_VAR_INIT(revolutionary_win, FALSE)
 		JOB_HEAD_OF_SECURITY,
 		JOB_PRISONER,
 		JOB_SECURITY_OFFICER,
+		JOB_CHAPLAIN,
```
Comment (nevimer) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:217]:
```suggestion
		JOB_CHAPLAIN, // BUBBER EDIT ADDITION
```

**Review by nevimer — APPROVED** [review]

**Comment by CADE-37** [comment]
Being frank, this was a terrible change. A consistent heretic and chaplain player I can confidently say chaplain is not a heretic counter.

On Skyrat Heretics are given metaprotections that dictate that no one except chaplains and curators can know what heretics are.

These protections do not exist on Bubber. Anyone is allowed to identify the use of magic and identify that it is a hostile action. Chaplains, on Bubber, have zero inherent advantages against heretics.

Being more frank, this change is stupid.

**Pull request merged**